core/Db/Schema/Mysql.php

```diff
@@ -102,6 +102,14 @@ public function getTablesCreateSql()
                             ) ENGINE=$engine DEFAULT CHARSET=utf8
             ",
 
+            'site_setting'    => "CREATE TABLE {$prefixTables}site_setting (
```

Comment:
It will be serialized for now yes. We might change it later but since it can store anything and we do need to keep the type of the `setting_value` for now. `json` etc would lose the type under circumstances eg `Array` vs `Objects`.

It doesn't store any instances there. Only `array`, `stdClass`, `int`, `float`, ...